docs/beta/plugins/discord.md

```diff
@@ -1,5 +1,248 @@
 ---
 title: Discord
+description: Setting up the Discord plugin.
 ---
 
-TODO
+The Discord plugin assigns Cred to contributors participating in Discord chats; for example, by posting and reacting to messages. 
+
+
+
+The plugin does this using a special bot that scans a Discord server and captures all messages and emoji reactions. The data is temporarily cached, but the
+source of truth for the Discord plugin is always the live
+Discord server. That means that if content is deleted on Discord, it will
+also disappear from the Discord plugin (after a cache refresh) {SB: copied from Discourse plugin, true for Discord also, right?}.
+
+
+For instructions on setting up the bot and configuring the plugin, see the [Configuration](#configuration) section below. 
+
+
+## Cred flow
+
+Cred is minted on Discord reactions. It then flows to connected nodes.
+
+TO DO: add example with diagram....The below diagram illustrates a typical pattern of Cred flow in Discord....
+
+
+
+The full set of node and edge types used by the Discord plugin are defined below.
+
+### Nodes
+
+The Discord plugin creates the following types of nodes in the contribution graph:
+
+- **User**:
+
+A Discord user account, e.g. [@decentralion {SB: correct link?}](). User accounts do not mint Cred, so setting a node weight would have no effect. Using the identity plugin, it's possible to "collapse" user nodes with other identity nodes into a single, canonical identity. For example, if a contributor had a Discord user account and a GitHub account, then the identity plugin can collapse those identities together.
+
+Users are connected to messages they author, to messages they react to, and to messages that mention them {SB: is this true?}.
+
+- **Guild Member**:
+
+A Discord user account, e.g. [@decentralion {SB: correct?}](), that has special powers. E.g. their reactions mint more Cred {SB: totally making this up here. Is this part of beta launch, or just a hidden alpha feature for MetaGame?}
+
+- **Bot**:
+
+{SB: below is copied from GitHub page. Still true for Discord? Do we support Discord bots?}
+A Discord user account that has been explicitly marked as a bot, via inclusion in bots.js. This is useful so we can filter out bot accounts from receiving grain or showing up in the Cred rankings.
+
+Bots have the same connections as users.
+
+- **Message**:
+
+A Discord message, e.g. ["Welcome to the SourceCred community!..."](https://discordapp.com/channels/453243919774253079/715770410955964477/715776215528505385) {SB: this is a link to the welcome message in the SourceCred Discord #start-here channel}. A message node will be connected to its author (user node creating the message), emoji reactions to the message, and any references to the message {SB: is this true??}. 
+
+[INSERT SCREENSHOT MESSAGE]
+
+- **Channel**:
```

Comment:
So channels arent actually their own nodes in the graph, they are just a part of the address structure for messages and reactions. e.g. for a message its something like `sourcecred/discord/message/channelid/messageid`

Not sure if this will change in the future though, but yea I would probably just change the writing here to reflect that its not a node. 

Something valuable to add here might be to list the address structure for each node type, that way people can know how the channels come into play.

Reply:
Valuable info if you're developing with the Discord plugin. Going to remove it for now though, as we're just listing actual nodes here and I want to get something from the user perspective up asap.

Reply:
I think this doc is largely going to be used by people installing and configuring the plugin. Not as much developers.  And it's already getting quite long. Suggest saving this info for more dev-facing docs.